test/suites/drift/suite_test.go

```diff
@@ -390,6 +397,11 @@ var _ = Describe("Drift", func() {
 		pod = env.EventuallyExpectHealthyPodCount(selector, numPods)[0]
 		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("ImageId", HaveValue(Equal(amdAMI))))
 
+		By("validating the deprecated status condition has been removed")
+		Eventually(func(g Gomega) {
+			g.Expect(nodeClass.StatusConditions().Get(v1.ConditionTypeAMIsDeprecated)).To(BeNil())
```

Comment:
We need to make sure we update the nodeClass from the API server or else we will not see a change in the nodecalss object. An example: https://github.com/aws/karpenter-provider-aws/blob/main/test/suites/expiration/suite_test.go#L141